### PR TITLE
fix doc errors flagged by newer sphinx versions

### PR DESCRIPTION
**Pull request by pkienzle**
See #407

### CODE REVIEW

File: sasmodels/compare.py [hunk 1]
```diff
@@ -872,7 +861,7 @@ def plot_profile(model_info, label='base', **args):
 
 
 def run_models(opts, verbose=False):
-    # type: (Dict[str, Any]) -> Dict[str, Any]
+    # type: (Dict[str, Any], bool) -> Dict[str, Any]
```
Comment (krzywon) [inline sasmodels/compare.py:864]: Optional[bool]?

File: sasmodels/compare.py [hunk 2]
```diff
@@ -949,9 +938,9 @@ def _print_stats(label, err):
 
 
 def plot_models(opts, result, limits=None, setnum=0):
-    # type: (Dict[str, Any], Dict[str, Any], Optional[Tuple[float, float]]) -> Tuple[float, float]
+    # type: (Dict[str, Any], Dict[str, Any], Optional[Tuple[float, float]], int) -> Tuple[float, float]
```
Comment (krzywon) [inline sasmodels/compare.py:941]: Optional[int]?

File: sasmodels/data.py [hunk 3]
```diff
@@ -39,15 +39,16 @@
...
+    OptLimits = Optional[Tuple[float, float]]
 except ImportError:
     pass
-else:
-    Data = Union["Data1D", "Data2D", "SesansData"]
 # pylint: enable=unused-import
 
 def load_data(filename, index=0):
-    # type: (str) -> Data
+    # type: (str, int) -> Data
```
Comment (krzywon) [inline sasmodels/data.py:51]: Optional[int]?

File: sasmodels/kernelpy.py [hunk 4]
```diff
@@ -200,16 +199,9 @@ def release(self):
...
-           form_radius,   # type: Callable[[], float]
-           nq,            # type: int
-           call_details,  # type: details.CallDetails
-           values,        # type: np.ndarray
-           cutoff,        # type: float
-          ):
-    # type: (...) -> None
+def _loops(parameters, form, form_volume, form_radius, nq, call_details,
+           values, cutoff):
+    # type: (np.ndarray, Callable([], np.ndarray], Callable[[], float], Callable[[], float], int, CallDetails, np.ndarray, float) -> None
```
Comment (krzywon) [inline sasmodels/kernelpy.py:204]: `Callable(` -> `Callable[`

File: doc/genapi.py [hunk 5]
```diff
@@ -78,6 +79,7 @@ def genfiles(package, package_name, modules, dir='api'):
     ('kernelpy', 'Python model evaluator'),
     ('list_pars', 'Identify all parameters in all models'),
     ('mixture', 'Mixture model evaluator'),
+    #('models.__init__', 'Builtin models'),
```
Comment (krzywon) [inline doc/genapi.py:82]: Are the built-in model docs being generated somewhere else?

File: sasmodels/compare.py (hunk 1 above)
Comment (pkienzle) [reply sasmodels/compare.py:864]: No. `Optional[bool]` would accept `None` as input. This is `bool` with a default value specified.

File: doc/genapi.py (hunk 5 above)
Comment (pkienzle) [reply doc/genapi.py:84]: Yes. I put a comment in to let maintainers know this.

**Pull request merged**